src/domain/index.ts

```diff
@@ -126,6 +126,18 @@ interface BridgeCommonFields {
   depositTxHash: string;
 }
 
+export type PendingBridge = {
+  status: "pending";
+  depositTxHash: BridgeCommonFields["depositTxHash"];
+  destinationAddress: BridgeCommonFields["destinationAddress"];
+  claimTxHash?: CompletedBridge["claimTxHash"];
+  from: BridgeCommonFields["from"];
+  to: BridgeCommonFields["to"];
+  token: BridgeCommonFields["token"];
+  amount: BridgeCommonFields["amount"];
+  fiatAmount: BridgeCommonFields["fiatAmount"];
+};
+
```

Comment:
```typescript
export type PendingBridge = Pick<
  BridgeCommonFields,
  "depositTxHash" | "destinationAddress" | "from" | "to" | "token" | "amount" | "fiatAmount"
> & {
  status: "pending";
  claimTxHash?: CompletedBridge["claimTxHash"];
};
```

I'm not a fan of `Pick` but in this case, it has the advantage of being less verbose and clearly spotting the actual differences between the two types: the `status` and the `claimTxHash` coming from the `CompletedBridge` with the optional modifier.

What do you think?

Reply:
I'm a big fan of this kind of this! I didn't add it before because I know that you hate this beautiful utils 🤣

But now that we are going to add it... I don't like to reference `claimTxHash` from an interface below its declaration, so I'd do this:

```
export type PendingBridge = Pick<
  BridgeCommonFields,
  "depositTxHash" | "destinationAddress" | "from" | "to" | "token" | "amount" | "fiatAmount"
> & {
  status: "pending";
  claimTxHash?: string;
};

export type CompletedBridge = BridgeCommonFields & {
  status: "completed";
  claimTxHash: NonNullable<PendingBridge["claimTxHash"]>;
};
```

Now we are using the beautiful things of TS 😎

Reply:
this `NonNullable<PendingBridge["claimTxHash"]>` hurts